### PR TITLE
EnNy OK

### PR DESCRIPTION
**Pull request by louist103**
Did some naming.  Might do some more. just wanted to get this in to get the conversation going.

### CODE REVIEW

**Comment by louist103** [comment]
Also didnt go OBJECT_NY due to some ZAP issues

File: src/overlays/actors/ovl_En_Ny/z_en_ny.h [hunk 1]
```diff
@@ -6,9 +6,31 @@
...
+    /* 0x0150 */ ColliderJntSph collider;
+    /* 0x0170 */ ColliderJntSphElement elements;
+    /* 0x01B0 */ char unk_1B0[0x18];
+    /* 0x01C8 */ s16 timer;
+    /* 0x01CA */ s16 unk_1CA;
+    /* 0x01CC */ s16 hitPlayer;
+    /* 0x01CE */ u16 unk_1CE;
+    /* 0x01D0 */ u8 unk_1D0;
+    /* 0x01D1 */ s8 unk_1D1;
+    /* 0x01D2 */ char unk_1D2[2];
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ny/z_en_ny.h:23]:
```suggestion
```

File: src/overlays/actors/ovl_En_Ny/z_en_ny.h [hunk 2]
```diff
@@ -6,9 +6,31 @@
...
 struct EnNy;
 
+typedef void (*EnNyActionFunc)(struct EnNy*, GlobalContext*);
+
 typedef struct EnNy {
     /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x16C];
+    /* 0x014C */ EnNyActionFunc actionFunc;
+    /* 0x0150 */ ColliderJntSph collider;
+    /* 0x0170 */ ColliderJntSphElement elements;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ny/z_en_ny.h:15]:
```suggestion
    /* 0x0170 */ ColliderJntSphElement colliderElements[1];
```

File: tools/overlayhelpers/DamageTable.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import struct
```
Comment (Roman971) [inline tools/overlayhelpers/DamageTable.py:1]: Was this file intended to be comitted?

File: weird cast.c [hunk 4]
```diff
@@ -0,0 +1,43 @@
+void EnNy_Init(Actor *thisx, GlobalContext *globalCtx) {
```
Comment (Roman971) [inline weird cast.c:1]: Same with this file

File: xmlcreate.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/python3
```
Comment (Roman971) [inline xmlcreate.py:1]: And this file

File: assets/xml/objects/object_ny.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<Root>
+    <File Name="object_ny" Segment="6">
+        <Texture Name="gEnNyRockBodyTex" Format="rgb5a1" Width="16" Height="16" Offset="0x0000"/>
+        <Texture Name="gEnNySpikeTex" Format="rgb5a1" Width="32" Height="32" Offset="0x0200"/>
+        <Texture Name="gEnNyMetalBodyTex" Format="rgb5a1" Width="32" Height="32" Offset="0x0A00"/>
```
Comment (Roman971) [inline assets/xml/objects/object_ny.xml:5]: Those textures need a name for the png they will generate.
For example:
```suggestion
        <Texture Name="gEnNyRockBodyTex" OutName="en_ny_rock_body" Format="rgb5a1" Width="16" Height="16" Offset="0x0000"/>
        <Texture Name="gEnNySpikeTex" OutName="en_ny_spike" Format="rgb5a1" Width="32" Height="32" Offset="0x0200"/>
        <Texture Name="gEnNyMetalBodyTex" OutName="en_ny_metal_body" Format="rgb5a1" Width="32" Height="32" Offset="0x0A00"/>
```

File: spec [hunk 7]
```diff
@@ -4783,7 +4782,9 @@ endseg
 beginseg
     name "object_ny"
     romalign 0x1000
-    include "build/baserom/object_ny.o"
+    //include "build/baserom/object_ny.o"
```
Comment (Roman971) [inline spec:4785]:
```suggestion
```

**Review by Roman971 — COMMENTED** [review]
Here are my suggestions

I couldn't make a comment for it but the .gitkeep you added in object_ny shouldn't be necessary.

File: src/overlays/actors/ovl_En_Ny/z_en_ny.c [hunk 8]
```diff
@@ -50,53 +61,516 @@ static ColliderJntSphInit D_80ABE354 = {
...
-    D_80ABE330,
+    sJntSphElementsInit,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ny/EnNy_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ny/EnNy_Destroy.s")
+static DamageTable sDamageTable = { {
+    0x00, 0x00, 0x00, 0xF2, 0x00, 0xF2, 0xF2, 0x12, 0x00, 0xF2, 0xF4, 0x24, 0xF2, 0xF2, 0xE4, 0x00,
+    0x00, 0x24, 0x00, 0x00, 0x00, 0x00, 0x00, 0xF4, 0xF2, 0x00, 0xF8, 0xF4, 0x00, 0x00, 0x00, 0x00,
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ny/z_en_ny.c:69]: I believe this should be updated to use the new format

File: src/overlays/actors/ovl_En_Ny/z_en_ny.c [hunk 9]
```diff
@@ -50,53 +61,516 @@ static ColliderJntSphInit D_80ABE354 = {
...
+    this->unk_1F0 = temp_f0;
+    this->actor.world.pos.y += temp_f0;
+
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    Actor_MoveForward(&this->actor);
+    Math_StepToF(&this->unk_1E4, this->unk_1E8, 0.1f);
+}
+
+static Vec3f sFireOffset[] = {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ny/z_en_ny.c:488]:
```suggestion
static Vec3f sFireOffsets[] = {
```

File: src/overlays/actors/ovl_En_Ny/z_en_ny.h [hunk 10]
```diff
@@ -6,9 +6,31 @@
...
+    /* 0x01D4 */ s32 unk_1D4;
+    /* 0x01D8 */ s32 unk_1D8;
+    /* 0x01DC */ s32 stoneTimer; // Delay for when to attempt to change forms
+    /* 0x01E0 */ f32 unk_1E0;
+    /* 0x01E4 */ f32 unk_1E4; // This, unk_1E8, and unk_1EC have to do with movement speed
+    /* 0x01E8 */ f32 unk_1E8;
+    /* 0x01EC */ f32 unk_1EC;
+    /* 0x01F0 */ f32 unk_1F0;
+    /* 0x01F4 */ f32 unk_1F4;
+    /* 0x01F8 */ Vec3f unk_1F8[0x10];
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ny/z_en_ny.h:33]:
```suggestion
    /* 0x01F8 */ Vec3f unk_1F8[16];
```

File: src/overlays/actors/ovl_En_Ny/z_en_ny.c [hunk 11]
```diff
@@ -50,53 +61,542 @@ static ColliderJntSphInit D_80ABE354 = {
...
+        // New initials
+        osSyncPrintf("ニュウ イニシャル[ %d ] ！！\n", this->actor.params);
+        this->actor.colChkInfo.mass = 0;
+        this->unk_1D4 = 0;
+        this->unk_1D8 = 0xFF;
+        this->unk_1E0 = 1.0f;
+        func_80ABCDBC(this);
+    } else {
+        //@Bug this block is never reached because params is never set to anything other than 0
+        // Dummy new initials
```
Comment (fig02) [inline src/overlays/actors/ovl_En_Ny/z_en_ny.c:140]: I wouldnt classify this as a bug, its just an unused param

Comment (fig02) [reply src/overlays/actors/ovl_En_Ny/z_en_ny.c:140]: a comment isnt even worth imo, but if you want to say something you can say "this mode is unused in the final game" or something

**Review by Roman971 — APPROVED** [review]

**Review by fig02 — APPROVED** [review]
